slahmr/run_rerun_vis.py

```diff
@@ -0,0 +1,331 @@
+    rr.set_time_sequence(f"frame_id_{phase_label}", 0)
+    rr.set_time_sequence("frame_id", 0)
+    rr.log_pinhole(
+        f"world/{phase_label}/camera/image",
+        child_from_parent=[[fx, 0, cx], [0, fy, cy], [0, 0, 1]],
+        width=width,
+        height=height,
+    )
+    rr.set_time_sequence(f"frame_id_{phase_label}", None)
+
+
+def log_phase_result(
+    cfg,
+    dataset: dataset.MultiPeopleDataset,
+    dev_id,
+    phase: str,
+    phase_label: str,
+    phase_result: dict,
+) -> None:
+    """Log results from one phase."""
+    B = len(dataset)
+    num_frames = dataset.seq_len
+    vis_mask = dataset.data_dict["vis_mask"]  # -1 out of frame, 0 occluded, 1 visible
+    device = get_device(dev_id)
+    phase_result = move_to(phase_result, device)
+
+    cfg = resolve_cfg_paths(cfg)
+    body_model, _ = load_smpl_body_model(cfg.paths.smpl, B * num_frames, device=device)
+
+    with torch.no_grad():
+        world_smpl = run_smpl(
+            body_model,
+            phase_result["trans"],
+            phase_result["root_orient"],
+            phase_result["pose_body"],
+            phase_result.get("betas", None),
+        )
+
+    # compute vertex normals on GPU
+    num_tracks = world_smpl["vertices"].shape[0]
+    num_meshes = num_tracks * num_frames
+    num_vertices = world_smpl["vertices"].shape[-2]
+    num_faces = world_smpl["faces"].shape[-2]
+
+    # faces has no batch dim here, use expand to avoid copy
+    meshes = pytorch3d.structures.Meshes(
+        verts=world_smpl["vertices"].reshape(num_meshes, num_vertices, 3),
+        faces=world_smpl["faces"].expand(num_meshes, num_faces, 3),
+    )
+
+    vertices = world_smpl["vertices"].numpy(force=True)
+    faces = world_smpl["faces"].numpy(force=True)
+    vertex_normals = (
+        meshes.verts_normals_padded()
+        .reshape(num_tracks, num_frames, num_vertices, 3)
+        .numpy(force=True)
+    )
+
+    # NOTE: if the meshes don't deform over time or are similar we could use the same
+    #  vertex normals for all frames and/or tracks
+
+    for frame_id in range(num_frames):
+        rr.set_time_sequence(f"frame_id_{phase_label}", frame_id)
+        rr.set_time_sequence("frame_id", frame_id)
+        translation = phase_result["cam_t"][1, frame_id].numpy(force=True)
+        rotation_mat = phase_result["cam_R"][1, frame_id].numpy(force=True)
+        rotation_q = transform.Rotation.from_matrix(rotation_mat).as_quat()
+        rr.log_rigid3(
+            f"world/{phase_label}/camera",
+            child_from_parent=(translation, rotation_q),
+            xyz="RDF",
+        )
+        for i, _ in enumerate(dataset.track_ids):
+            if vis_mask[i][frame_id] >= 0:
+                rr.log_mesh(
+                    f"world/{phase_label}/#{i}",
+                    vertices[i, frame_id],
+                    indices=faces,
+                    normals=vertex_normals[i, frame_id],
+                    albedo_factor=_index_to_color(i),
```

Comment:
Could open an issue here for letting `AnnotationContext` apply to meshes as well

Reply:
will do